cards.js

```diff
@@ -17,7 +17,46 @@ function shuffle(array){
 }
 
 function buildDeck(){
+  var deck = [];
+    for (i = 0; i < values.length; i++){
+      newCard = {suit: "Clubs", value: values[i]};
+      deck.push(newCard);
+      }
+    for (i = 0; i < values.length; i++){
+      newCard = {suit: "Diamonds", value: values[i]};
+      deck.push(newCard);
+        }
+    for (i = 0; i < values.length; i++){
+      newCard = {suit: "Hearts", value: values[i]};
+      deck.push(newCard);
+        }
+    for (i = 0; i < values.length; i++){
+      newCard = {suit: "Spades", value: values[i]};
+      deck.push(newCard);
+        }
```

Comment:
While there's nothing wrong with this code, I think a nested loop (like we did in the ruby solution) would be generally preferable. (It's less code, and I think it reads a bit easier, but you could definitely argue that point!)